core/src/main/java/org/apache/carbondata/core/scan/filter/executer/MeasureColumnExecuterFilterInfo.java

```diff
@@ -16,15 +16,143 @@
  */
 package org.apache.carbondata.core.scan.filter.executer;
 
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.carbondata.core.metadata.datatype.DataType;
+import org.apache.carbondata.core.metadata.datatype.DataTypes;
+
+import it.unimi.dsi.fastutil.booleans.BooleanOpenHashSet;
+import it.unimi.dsi.fastutil.bytes.ByteOpenHashSet;
+import it.unimi.dsi.fastutil.doubles.DoubleOpenHashSet;
+import it.unimi.dsi.fastutil.floats.FloatOpenHashSet;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+import it.unimi.dsi.fastutil.shorts.ShortOpenHashSet;
+
+/**
+ * Below class will be used to keep all the filter values based on data type
+ * for measure column.
+ * In this class there are multiple type of set is used to avoid conversion of
+ * primitive type to primitive object to avoid gc which cause perofrmace degrade when
+ * number of records are high
+ */
 public class MeasureColumnExecuterFilterInfo {
 
   Object[] filterKeys;
 
-  public void setFilterKeys(Object[] filterKeys) {
+  private ByteOpenHashSet byteOpenHashSet;
+
+  private IntOpenHashSet intOpenHashSet;
+
+  private DoubleOpenHashSet doubleOpenHashSet;
+
+  private ShortOpenHashSet shortOpenHashSet;
+
+  private Set<Object> bigDecimalHashSet;
+
+  private LongOpenHashSet longOpenHashSet;
+
+  private BooleanOpenHashSet booleanOpenHashSet;
+
+  private FloatOpenHashSet floatOpenHashSet;
```

Comment:
I feel this many fileds are not necessary, just take one field with type of super class/interface. During reading you can typecast it.